internal/services/springcloud/spring_cloud_service_resource.go

```diff
@@ -202,8 +203,18 @@ func resourceSpringCloudService() *pluginsdk.Resource {
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
 						"instrumentation_key": {
-							Type:     pluginsdk.TypeString,
-							Required: true,
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							Deprecated:   "Deprecated in favour of `connection_string` which is the connection string for Application Insights",
```

Comment:
lets be clear here then
```suggestion
							Deprecated:   "This property is due to be removed from the service API and thus hs been deprecated. Please switch to using the `connection_string` property with the connection string for the Application Insights instance.",
```